packages/core/src/lib/code-editors/monaco.ts

```diff
@@ -0,0 +1,70 @@
+import type monaco from 'monaco-editor';
+import { writable } from 'svelte/store';
+
+export type Monaco = typeof monaco;
+export type MonacoSupportedThemes = 'vs' | 'vs-dark';
+export type MonacoMarker = monaco.editor.IMarker;
+
+export interface MonacoOptionsParameters {
+  value: string;
+  language: string;
+  theme: MonacoSupportedThemes;
+  minimap: boolean;
+}
+
+export const useMonaco = () => {
+  const instance = writable<Monaco>();
+  const initialized = writable(false);
```

Comment:
The `initialized` writable seems redundant with the `instance` writable, which will be `undefined` prior to initialization. Also, could we move the worker setup here, too, to make this closer to [monaco's vite example](https://github.com/microsoft/monaco-editor/blob/main/docs/integrate-esm.md#using-vite)

Reply:
I can try, this is a departure from the link @micheal-parks sent. This one imports from the root `monaco-editor` package where the other imports from `monaco-editor/esm/vs/editor/editor.api`. I tried importing from `monaco-editor/esm/vs/editor/editor.api` but it does not work despite it not throwing any type errors.